### ISSUE TITLE
Refactor into MVC, with a model for and unit tests.

### ISSUE THREAD

**Issue by pconrad** [UCSB-CS56-Projects/cs56-games-simon-says#9]
Imagine a model for the simon says game.

Perhaps the model is a SimonSaysGame object.    The constructor might initially create a sequence of one color in length.    It might then have a method getCurrentSequence that returns a sequence of colors (maybe as a string? maybe as an ArrayList of colors or letters or ints representing colors?)

Maybe then there is a method startTurn and a method guessNextColor that is called repeatedly, and returns either true if the color was guessed correctly, or false if the color was guessed incorrectly, and an endTurn method that might be invoked automatically after a timer expires to indicate the turn is over (if there is a time limit)      There might also be a method "getSequenceLength" that returns the length of the sequence currently being guessed, and a sequenceComplete() method that indicates if the user already guessed the entire sequence correctly and its time for a new set of guesses to start.

In short, come up with a model, and a set of unit tests that could be invoked on that model.     

Put the code in this directory into one more package level down, namely: 

src.edu.ucsb.cs56.projects.games.simon_says.view

then make a new class SimonSaysGame under 

src.edu.ucsb.cs56.projects.games.simon_says.model

along with SimonSaysTest

and use TDD to make the tests pass.   For THIS issue to get full points, it is enough to have a plausible set of methods on the the SimonSaysGame object, along with unit tests that could be used to replace the game logic in the current SimonSays game code under view...but integrating that with the GUI is a separate issue.

~estimated 300
